src/components/calendar/bl-calendar.ts

```diff
@@ -0,0 +1,545 @@
+      this._selectedRangeDates.startDate = calendarDate;
+      this._selectedDates.push(calendarDate);
+    } else if (this._selectedRangeDates.startDate && !this._selectedRangeDates.endDate) {
+      if (calendarDate.getTime() > this._selectedRangeDates.startDate.getTime()) {
+        this._selectedRangeDates.endDate = calendarDate;
+        this._selectedDates.push(calendarDate);
+      } else if (calendarDate.getTime() < this._selectedRangeDates.startDate.getTime()) {
+        const temp = this._selectedRangeDates.startDate;
+
+        this._selectedRangeDates.startDate = calendarDate;
+        this._selectedRangeDates.endDate = temp;
+        this._selectedDates.splice(
+          0,
+          this._selectedDates.length,
+          this._selectedRangeDates.startDate,
+          this._selectedRangeDates.endDate
+        );
+      }
+    } else if (this._selectedRangeDates.startDate && this._selectedRangeDates.endDate) {
+      this._selectedRangeDates.startDate = calendarDate;
+      this._selectedRangeDates.endDate = undefined;
+      this._selectedDates.splice(0, this._selectedDates.length, this._selectedRangeDates.startDate);
+    }
+    this.setHoverClass();
+  }
+
+  checkIfSelectedDate(calendarDate: CalendarDate) {
+    const day = this._selectedDates.find(selectedDate => {
+      return calendarDate.getTime() === selectedDate.getTime();
+    });
+
+    return !!day;
+  }
+  checkIfDateIsToday(calendarDate: CalendarDate) {
+    const today = new Date();
+
+    return (
+      today.getDate() === calendarDate.getDate() &&
+      today.getMonth() === calendarDate.getMonth() &&
+      today.getFullYear() === calendarDate.getFullYear()
+    );
+  }
+  checkIfDateIsDisabled(calendarDate: CalendarDate) {
+    if (
+      calendarDate.getTime() < this.minDate?.getTime() ||
+      calendarDate.getTime() > this.maxDate?.getTime()
+    ) {
+      return true;
+    }
+
+    if (Array.isArray(this.disabledDates)) {
+      const day = this.disabledDates.find(disabledDate => {
+        return calendarDate.getTime() === new Date(disabledDate).getTime();
+      });
+
+      return !!day;
+    }
+    return false;
+  }
+
+  setHoverClass() {
+    this.clearRangePickerStyles();
+
+    if (this._selectedRangeDates.startDate && this._selectedRangeDates.endDate) {
+      setTimeout(() => {
+        const startDateParentElement = this.shadowRoot?.getElementById(
+          `${this._selectedRangeDates.startDate?.getTime()}`
+        )?.parentElement;
+
+        startDateParentElement?.classList.add("range-start-day");
+
+        const endDateParentElement = this.shadowRoot?.getElementById(
+          `${this._selectedRangeDates.endDate?.getTime()}`
+        )?.parentElement;
+
+        endDateParentElement?.classList.add("range-end-day");
+        const rangeDays = [...this.createCalendarDays().values()].flat().filter(
+          date =>
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-expect-error
```

Comment:
Can we remove them and find another way to solve it? Let me know if you need assistance.

Reply:
can you check if the solution is good ?